numba/tests/test_array_methods.py

```diff
@@ -779,18 +787,198 @@ def test_sum(self):
+             np.ones(1, arr_dtype),
+             np.ones((7, 3), arr_dtype) * -5]
+            for arr_dtype in all_dtypes]
+
+        for arr_list in all_test_arrays:
+            for arr in arr_list:
+                for axis in (0, 1, 2):
+                    if axis > len(arr.shape)-1:
+                        continue
+                    with self.subTest("Testing np.sum(axis) with {} "
+                                      "input ".format(arr.dtype)):
+                        npy_res = pyfunc(arr, axis=axis)
+                        numba_res = cfunc(arr, axis=axis)
+                        if isinstance(numba_res, np.ndarray):
+                            self.assertPreciseEqual(
+                                npy_res.astype(out_dtypes[arr.dtype]),
+                                numba_res.astype(out_dtypes[arr.dtype]))
+                        else:
+                            # the results are scalars
+                            self.assertEqual(npy_res, numba_res)
+
+    def test_sum_dtype_kws(self):
+        pyfunc = array_sum_dtype_kws
+        cfunc = jit(nopython=True)(pyfunc)
+        all_dtypes = [np.float64, np.float32, np.int64, np.int32, np.uint32,
+                      np.uint64, np.complex64, np.complex128]
+        # timedelta test cannot be enabled until issue #4540 is fixed
+        # all_dtypes += [np.timedelta64]
+        all_test_arrays = [
+            [np.ones((7, 6, 5, 4, 3), arr_dtype),
+             np.ones(1, arr_dtype),
+             np.ones((7, 3), arr_dtype) * -5]
+            for arr_dtype in all_dtypes]
+
+        out_dtypes = {np.dtype('float64'): [np.float64],
+                      np.dtype('float32'): [np.float64, np.float32],
+                      np.dtype('int64'): [np.float64, np.int64, np.float32],
+                      np.dtype('int32'): [np.float64, np.int64, np.float32, np.int32],
+                      np.dtype('uint32'): [np.float64, np.int64, np.float32],
+                      np.dtype('uint64'): [np.float64, np.int64],
+                      np.dtype('complex64'): [np.complex64],
+                      np.dtype('complex128'): [np.complex128],
+                      np.dtype('timedelta64'): [np.timedelta64]}
+
+        for arr_list in all_test_arrays:
+            for arr in arr_list:
+                for out_dtype in out_dtypes[arr.dtype]:
+                    subtest_str = ("Testing np.sum with {} input and {} output"
+                                   .format(arr.dtype, out_dtype))
+                    with self.subTest(subtest_str):
+                        self.assertPreciseEqual(pyfunc(arr, dtype=out_dtype),
+                                                cfunc(arr, dtype=out_dtype))
+
+    def test_sum_dtype_kws_negative(self):
+        pyfunc = array_sum_dtype_kws
+        cfunc = jit(nopython=True)(pyfunc)
+        dtype = np.float64
+        # OK
+        a = np.ones((7, 6, 5, 4, 3))
+        self.assertFalse(type(pyfunc(a, dtype=np.int32)) == cfunc(a, dtype=dtype))
```

Comment:
This is comparing a type to a value, what is the intention here?

Reply:
I think the intention (with a mistake) was to compare types but it must have been an early test. the main tests are much better now, this test does not add anything. I'm deleting it.

Reply:
thanks